dev/com.ibm.ws.org.apache.cxf.cxf.rt.transports.http.3.2/src/org/apache/cxf/transport/http/URLConnectionHTTPConduit.java

```diff
@@ -472,6 +476,24 @@ protected boolean usingProxy() {
         protected void setFixedLengthStreamingMode(int i) {
             // [CXF-6227] do not call connection.setFixedLengthStreamingMode(i)
             // to prevent https://bugs.openjdk.java.net/browse/JDK-8044726
+        
+            // Liberty Change for issue #25866-unexpected_EOF_from_server
+            // The bug [CXF-6227] is a performance bug observed for Java 7
+            // In case of a performance drop is observed, need to create a work around
+            boolean isRestMessage = true; // It's safer to set isRestMessage, because 
+            // calling setFixedLengthStreamingMode method creates IllegalStateException at JAX-RS 
+            Exchange exchange = outMessage.getExchange();
+            if(exchange != null)        {
+                //This property return true if it's a rest web service
+                isRestMessage = PropertyUtils.isTrue(exchange.get(org.apache.cxf.message.Message.REST_MESSAGE));
+            }
+            // Don't call setFixedLengthStreamingMode for JAX-RS and 
+            // when cxf.set.fixedlengthstreaming not equal to cxf.set.fixedlengthstreaming
+            // This property is a safe measure in case enabling this property cause an issue
+            if(!isRestMessage && PropertyUtils.isTrue(("cxf.set.fixedlengthstreaming"))  )   { 
```

Comment:
This will not work, the system property needs to be checked using System.getProperty("cxf.set.fixedlengthstreaming"), in the starting static block.